File: app/src/main/java/antoninovitale/dropcodechallenge/api/PunkAPI.java
package antoninovitale.dropcodechallenge.api;

import java.util.List;

import antoninovitale.dropcodechallenge.api.model.Beer;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by antoninovitale on 28/08/2017.
 */
public interface PunkAPI {
    String API_BASE_URL = "https://api.punkapi.com";

    @GET("v2/beers/random")
    Call<List<Beer>> getRandomBeer();

    @GET("v2/beers")
    Call<List<Beer>> getBeers(@Query("page") String page, @Query("per_page") String count);

}

File: app/src/main/java/antoninovitale/punkapi/app/api/PunkAPI.kt
package antoninovitale.punkapi.app.api

import antoninovitale.punkapi.app.api.model.Beer
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Created by antoninovitale on 28/08/2017.
 */
interface PunkAPI {
    @get:GET("v2/beers/random")
    val randomBeer: Call<List<Beer?>?>?

    @GET("v2/beers")
    fun getBeers(
            @Query("page") page: String?,
            @Query("per_page") count: String?): Call<List<Beer?>?>?

    companion object {
        const val API_BASE_URL = "https://api.punkapi.com"
    }
}